manifests/init.pp

```diff
@@ -30,9 +33,9 @@
 class foreman_scap_client(
   $server,
   $port,
-  $ca_file          = '/var/lib/puppet/ssl/certs/ca.pem',
-  $host_certificate = "/var/lib/puppet/ssl/certs/${fqdn}.pem",
-  $host_private_key = "/var/lib/puppet/ssl/private_keys/${fqdn}.pem",
+  $ca_file          = get_ca_cert(),
```

Comment:
I think these can be implemented using the params pattern:

```puppet
class foreman_scap_client(
  $server,
  $port,
  $ca_file          = $::foreman_scap_client::params::ca_file,
  $host_certificate = $::foreman_scap_client::params::host_certificate,
  $host_private_key = $::foreman_scap_client::params::host_private_key,
  $policies,
) inherits foreman_scap_client::params {
}
```

Then in params.pp something like:
```puppet
class foreman_scap_client::params {
  $ca_file          = pick($::rh_certificate_consumer_ca_file, '/var/lib/puppet/ssl/certs/ca.pem')
  $host_certificate = pick($::rh_certificate_consumer_host_cert, "/var/lib/puppet/ssl/certs/${fqdn}.pem")
  $host_private_key = pick($::rh_certificate_consumer_host_key, "/var/lib/puppet/ssl/private_keys/${fqdn}.pem")
}
```

(pick comes from puppetlabs-stdlib, but you can also implement is using simple if logic)